### PR TITLE
Packaging Cake.Common.dll into Cake.Common.nupkg.

### PR DESCRIPTION
**Pull request by olivier-spinelli**
Hello everybody,

A very small PR that packages Cake.Common into a .nupkg. This does not change anything to the project except that Cake.Common becomes usable from an autonomous application that may need it (once it will be pushed on nuget.org of course!).

### CODE REVIEW

**Comment by RichiCoder1** [comment]
Is there any particular reason you'd like Cake.Common available as a package?

**Comment by patriksvensson** [comment]
Before doing something like this we should add support to add dependencies dynamically to a nuget specification. If we don't, we will forget to update the version numbers in every release.

File: nuspec/Cake.Common.nuspec [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    <description>The Cake common library.</description>
+    <summary>Cake (C# Make) is a build automation system with a C# DSL to do things like compiling code, copy files/folders, running unit tests, compress files and build NuGet packages.</summary>
+    <licenseUrl>https://github.com/cake-build/cake/blob/develop/LICENSE</licenseUrl>
+    <projectUrl>https://github.com/cake-build/cake</projectUrl>
+    <iconUrl>https://raw.githubusercontent.com/cake-build/graphics/master/png/cake-medium.png</iconUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <copyright>Copyright (c) Patrik Svensson, Mattias Karlsson and contributors</copyright>
+    <tags>Cake Script Build</tags>
+    <dependencies>
+      <dependency id="Cake.Core" version="0.5.2" />
```
Comment (patriksvensson) [inline nuspec/Cake.Common.nuspec]: Ok, could actually write `<dependency id="Cake.Core" version="$version$" />` to get this to work.

File: build.ps1 [hunk 2]
```diff
@@ -62,4 +62,5 @@ if (!(Test-Path $CAKE_EXE)) {
 
 # Start Cake
 Invoke-Expression "$CAKE_EXE `"$Script`" -target=`"$Target`" -configuration=`"$Configuration`" -verbosity=`"$Verbosity`" $UseMono $UseDryRun $UseExperimental"
+pause
```
Comment (patriksvensson) [inline build.ps1]: No pause here.

File: src/Cake.sln [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 2013
-VisualStudioVersion = 12.0.31101.0
+# Visual Studio 14
```
Comment (patriksvensson) [inline src/Cake.sln]: We're using VS2013 at the moment.

File: nuspec/Cake.Common.nuspec [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+  <metadata xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+    <id>Cake.Common</id>
+    <version>0.0.0</version>
+    <authors>Patrik Svensson</authors>
+    <owners>Patrik Svensson</owners>
+    <description>The Cake common library.</description>
```
Comment (patriksvensson) [inline nuspec/Cake.Common.nuspec]: We would need a better description here.

**Comment by patriksvensson** [comment]
@olivier-spinelli Left some comments that would allow this PR to be merged.

**Comment by olivier-spinelli** [comment]
Thanks!
I' mo doing it right now.

**Comment by olivier-spinelli** [comment]
Since the nuget's summary must be short and the description longer, here is my proposal for Cake.Common:

    <summary>Provides common features to Cake build scripts.</summary>
    <description>
    Provides aliases (extension methods on Cake context) that support CI, build, unit tests, zip, signing, etc. for Cake.  
    Cake (C# Make) is a build automation system with a C# DSL to do things like compiling code, copy files/folders, running unit tests, compress files and build NuGet packages.
    </description>

Are you okay with this?

**Comment by patriksvensson** [comment]
I think we should keep the summary like it is and change everything later on. Need to check with @devlead as well about this.

```xml
<summary>Cake (C# Make) is a build automation system with a C# DSL to do things like compiling code, copy files/folders, running unit tests, compress files and build NuGet packages.</summary>
<description>Provides aliases (extension methods on Cake context) that support CI, build, unit tests, zip, signing, etc. for Cake.</description>
```

**Comment by olivier-spinelli** [comment]
OK! I include your change in the PR. (Please note that according to https://docs.nuget.org/create/nuspec-reference, <description> is a required long text and <summary> an optional short one.)

**Comment by devlead** [comment]
@olivier-spinelli thanks for your contribution! This is now merged and will come with next release.

**Comment by olivier-spinelli** [comment]
Thanks for having accepted it!
I just want to notice you that I did not modify the *Publish-MyGet* target in build.cake since I was not sure what to do here. May I have added those lines?
```csharp
    // Get the path to Cake.Common package.
    var commonPackage = nugetRoot + File("Cake.Common." + semVersion + ".nupkg");

    // Push the package.
    NuGetPush(commonPackage, new NuGetPushSettings {
        Source = "https://www.myget.org/F/cake/api/v2/package",
        ApiKey = apiKey
    });
```

**Comment by patriksvensson** [comment]
@olivier-spinelli No. Only Cake is distributed via MyGet at the moment. Not against adding Cake.Core and Cake.Common there as well though. @devlead what do you think?

**Comment by devlead** [comment]
I have no issue with adding these to NuGet.
Only important thing is that they're not included in any addin/3rd party Nuget packages as that may conflict / result in issues within cake script as those are intended to be distributed with the "runtime".

**Comment by olivier-spinelli** [comment]
Yes. This Cake.Common package must only be used by other hosts than the existing runtime. Typically by CodeCake I'm working on. Thanks for your help in this!
Do you know when this will be available? (Cake 0.5.3?)

**Comment by patriksvensson** [comment]
You can check out the mile stone for v0.5.3 under issues on the github page. Not sure when this will be, but I'll sync with @devlead about this.

**Comment by devlead** [comment]
Mewanwhile @olivier-spinelli  PreRelease versions are now available on the MyGet feed `https://www.myget.org/F/cake` which is publicly available.
```powershell
:> nuget list -Source https://www.myget.org/F/cake -PreRelease
Cake 0.5.2-build-712
Cake.Common 0.5.2-build-712
Cake.Core 0.5.2-build-712
```
This is the feed we use for pre release testing, so no guarantees there. But that said it always contains latest merged code that builds.

**Comment by olivier-spinelli** [comment]
Thanks a lot, I'try to give it a try asap.

**Comment by jrnail23** [comment]
@devlead, if the currently merged changes are targeted for 0.5.3 why not increment those pre-release packages to 0.5.3-build-[buildNumber]?  That's more in line with SemVer expectations.

**Comment by patriksvensson** [comment]
Yes. We (I) need to be better at updating the version number after an release. Will fix when I have access to a computer.

**Comment by devlead** [comment]
@jrnail23 yes, basically just need to update ReleaseNotes.md, done that now so new packages should arrive shortly.

We should probably do that post every release, adding something like bellow to [ReleaseNotes.md](https://github.com/cake-build/cake/blob/develop/ReleaseNotes.md), would probably be sufficient:
```
### New in 0.5.4 (Pre-release)
* Pre-release
```

**Comment by devlead** [comment]
@jrnail23 & @patriksvensson  0.5.3 pre release packages are now on MyGet
```powershell
> nuget list -Source https://www.myget.org/F/cake -PreRelease
Cake 0.5.3-build-713
Cake.Common 0.5.3-build-713
Cake.Core 0.5.3-build-713
```

**Comment by jrnail23** [comment]
:+1:

**Comment by patriksvensson** [comment]
Perfect and I agree. We neee a relase checklist soon and we need to make it more automated :)

**Comment by jrnail23** [comment]
If only there was a tool we could use to automate such build tasks... perhaps one that uses Roslyn to enable C# scripting? Oh, and its should also rhyme with "make" :stuck_out_tongue_winking_eye:

**Comment by devlead** [comment]
we should indeed Cake it...🍰😎

**Comment by RichiCoder1** [comment]
Might I recommend Particular Labs GitVersion? We use it at my work and it's great for taking care of versioning for you.

**Comment by olivier-spinelli** [comment]
That's what I'm trying to do. Everything is driven by the repository and the semver tags you put on commit points. 
I tried GitVersion before... I'm trying to build a simpler alternative that does not rely on a strict observance of branches but purely on the code base (SHA1 of tree) and parent/child topology only (respecting/enforcing a XXXflow must be an option up to me).

All this relies on a more operational semantic versioning (see http://csemver.org).
It's not ready... but all of this is here: https://github.com/SimpleGitVersion.

**Comment by devlead** [comment]
Good points:+1: Currently it's just a line in a text file so it's not too complicated though.

**Pull request merged**